components/net/fetch/methods.rs

```diff
@@ -512,9 +534,24 @@ fn scheme_fetch(request: &mut Request,
                                         }
                                         let length = {
                                             let mut buffer = reader.fill_buf().unwrap().to_vec();
-                                            let buffer_len = buffer.len();
+                                            let mut buffer_len = buffer.len();
                                             if let ResponseBody::Receiving(ref mut body) = *res_body.lock().unwrap() {
-                                                body.extend_from_slice(&buffer);
+                                                let offset = usize::min({
+                                                    if let Some(end) = end {
+                                                        let remaining_bytes = end as usize - body.len();
+                                                        if remaining_bytes <= FILE_CHUNK_SIZE {
+                                                            // This is the last chunk so we set buffer len to 0 to break
+                                                            // the reading loop.
+                                                            buffer_len = 0;
```

Comment:
This affects the usize::min call, right? We won't append the final chunk to the body in this case.

Reply:
I believe it does not affect it. `usize::min` gets `buffer.len()` as the second param, not `buffer_len`.